homework/index.js

```diff
@@ -16,32 +20,157 @@
       } else {
-        elem.setAttribute(key, value);
+        createAndAppend('p', parent, {
+          text: 'Contributions',
+          class: 'contributors__header',
+        });
+        const ul = createAndAppend('ul', parent, { class: 'contributors__list' });
+        data.forEach(contributor => {
+          const li = createAndAppend('li', ul, { class: 'contributors__list-item' });
+          createAndAppend('img', li, {
+            src: contributor.avatar_url,
+            alt: contributor.login,
+            class: 'contributors__avatar',
+          });
+          const contributorData = createAndAppend('div', li, { class: 'contributors__info' });
+          createAndAppend('div', contributorData, {
+            text: contributor.login,
+            class: 'contributors__name',
+          });
+          createAndAppend('div', contributorData, {
+            text: contributor.contributions,
+            class: 'contributors__label',
+          });
+          li.addEventListener('click', () => {
+            window.open(contributor.html_url);
+          });
+        });
       }
     });
-    return elem;
+  }
+
+  function createStaticComponents() {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('h3', header, {
+      class: 'header__title',
+      text: 'HYF Repositories',
+    });
+    const repositorySelect = createAndAppend('select', header, {
+      'aria-label': 'repository-select',
+      class: 'header__select',
+      id: 'repository-select',
+    });
+    const container = createAndAppend('main', root, { class: 'container' });
+    const detailsComponent = createAndAppend('div', container, { class: 'details card' });
+    const detailsTable = createAndAppend('table', detailsComponent);
+    const contributorsComponent = createAndAppend('div', container, {
+      class: 'contributors card',
+    });
+
+    return {
+      root,
+      repositorySelect,
+      detailsTable,
+      contributorsComponent,
+    };
   }
 
   function main(url) {
+    const {
+      root,
+      repositorySelect,
+      detailsTable,
+      contributorsComponent,
+    } = createStaticComponents();
+
     fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const sortedData = data.sort((a, b) =>
+          a.name.toLowerCase().localeCompare(b.name.toLowerCase()),
+        );
+        const repositories = sortedData.map(repository => ({
+          name: repository.name,
+          description: repository.description,
+          forks: repository.forks,
+          updated: repository.updated_at,
+          repositoryUrl: repository.html_url,
+          contributorsUrl: repository.contributors_url,
+        }));
```

Comment:
I saw this similar approach in the homework of @suh3yb. Did you guys work together?

Reply:
We didn't wotk together.

I tried many different approaches in order to get the expected data. My first approach was create components from an object model. I could create static components but I couldn't pass data to dinamic parts. It was totally different so I gave up after three days and turned back the normal way.

I've viewed some of the other works, also from the previous classes, in order to get inspired. That's why it's similar to one.